### ISSUE TITLE
feat: Uygulamaya İngilizce - Türkçe dil desteği eklenmeli

### ISSUE THREAD

**Issue by ahmettulutas** [acikkaynak/deprem-yardim-frontend#587]
## İsteğiniz şeyi tanımlayın 
Uygulamaya react-intl ile bir dil desteği getirilmelidir.

-- bu bilgiyi ekleyiniz
** discord kullanıcı adı: @afetharita#0001 **

## Değerlendirdiğiniz alternatifleri tanımlayın
Değerlendirdiğiniz herhangi bir alternatif çözüm veya özellik hakkında açık ve kısa bir açıklama.

## Ek Bağlam
Bu özellik isteği hakkında herhangi bir başka bağlam veya ekran görüntüsü ekleyin.

**Comment by sarpkayature** [comment]
@ahmettulutas i18n implemente edilebilir.

**Comment by emr3rden** [comment]
Unchecked etiketini kaldırdım fakat template'i incelersen eğer ona göre discord kullanıcı adını doğru girersen çok daha iyi olur. Template'in linki: https://github.com/acikkaynak/deprem-yardim-frontend/blob/development/.github/ISSUE_TEMPLATE/feature_request.md

**Comment by canerakin111** [comment]
@ahmettulutas React'in dilini mi ingilizce olmasini teklif ediyorsun? Biraz daha detay alabilir miyiz?
@eraygundogmus buraya bir bakabilir misin?

**Comment by ahmettulutas** [comment]
> @ahmettulutas React'in dilini mi ingilizce olmasini teklif ediyorsun? Biraz daha detay alabilir miyiz? @eraygundogmus buraya bir bakabilir misin?

Hayır. Bu uygulamaya bir dil özelliği ekleyebilirim. Böylece sadece türkçe bilen değil yabancı kullanıcılar da bu uygulamayı kullanabilir. Hatta geliştirme hazır. Onaylarsanız pr atabilirim. https://vimeo.com/797375383/2b87da0f33

**Comment by ahmettulutas** [comment]
Aslında 280px width'e kadar bu buton çalışır. Yani bu butonu mobilde kullanamama gibi bir durum söz konusu değil. Cihaz dilini yakalayarak verilebilir tabii ancak cihaz dili ingilizce dışında bir şey ise fallback hep ingilizce olur. Aksi takdirde bütün cihaz dillerine yönelik translation dosyaları oluşturmak gerekir. Ben bunu default language Türkçe ve yabancı bir kullanıcı dilerse kullanabilsin, İngilizce dil seçeneği olsun ve Türkçe'den İngilizce'ye switch edebilsin diye ekledim. Pratik ve hızlı oldu.

**Comment by ahmettulutas** [comment]
> @ahmettulutas Yok dediğini anladım fakat mobile de 280px üstü ols abile alan daraltacak, kullanımı kolaylaştırmamız gerekiyor daha çok mobile için. Bu sebeple cihaz dilini yakalamayı önerdim, ayrıca tüm dilleri yapmamıza gerek yok, üstte de açıklamıştım TR ise TR, TR olmayan her cihaz dili için EN gösterilir.
> 
> Bunu product'a veya lead'lere bırakabiliriz, benimki kullanıcı arayüzünü basit tutmak için bir öneri sadece.

Ben next.js projesi olduğu için next'in useRouter() hookunu ve react-intl kütüphanelerini kullanarak conventional bir çözüm buldum açıkçası. Butonu mobilde alan kaplamaması için küçültebilirim.
